Development/ssl/ssl_utils.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SSL_SSL_UTILS_H
+#define SSL_SSL_UTILS_H
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <openssl/x509v3.h> // for X509V3_EXT_conf_nid
```

Comment:
Hmm, `X509V3_EXT_conf_nid` doesn't appear in this header file, so that can't be it... `X509_free` maybe??